src/gvpn_controller.py

```diff
@@ -1,22 +1,23 @@
 #!/usr/bin/env python
 
 from ipoplib import *
-
+# GvpnUdpServer inherits the UdpServer class of the core IPOP Library
 class GvpnUdpServer(UdpServer):
     def __init__(self, user, password, host, ip4):
         UdpServer.__init__(self, user, password, host, ip4)
         self.idle_peers = {}
-        self.user = user
-        self.password = password
-        self.host = host
+        self.user = user # XMPP username
+        self.password = password # XMPP Password
+        self.host = host # XMPP host
         self.ip4 = ip4
-        self.uid = gen_uid(ip4)
+        self.uid = gen_uid(ip4) # UID is the SHA-1 hash of the IPv4 address.
```

Comment:
This comment is a bit verbose, I think the best way to write it is to distill it down the the absolute minimum needed to gain context, so instead, I would just say something like

```python
# SHA-1 hash
```